src/main/java/com/google/devtools/build/lib/skyframe/WatchServiceDiffAwareness.java

```diff
@@ -115,7 +116,23 @@ public View getCurrentView(OptionsProvider options) throws BrokenDiffAwarenessEx
     Set<Path> modifiedAbsolutePaths;
     if (isFirstCall()) {
       try {
-        registerSubDirectoriesAndReturnContents(watchRootPath);
+        // Due to a known issue nested watches may result in errors on windows: https://bugs.openjdk.java.net/browse/JDK-6972833
+        // Therefore on windows we register using the special ExtendedWatchEventModifier.FILE_TREE
+        // This watches a folder recursively so there is no need to apply this ourselves
+        if(OS.getCurrent() == OS.WINDOWS) {
```

Comment:
It would be good to call OS.getCurrent() once in the constructor, store it in a field, and then use that here.